Sources/TuistCore/Simulator/SimulatorController.swift

```diff
@@ -97,13 +97,22 @@ public final class SimulatorController: SimulatorControlling {
     private let jsonDecoder = JSONDecoder()
     private let userInputReader: UserInputReading
 
-    public init(userInputReader: UserInputReading = UserInputReader()) {
+    private let system: Systeming
+    private let devEnvironment: DeveloperEnvironmenting
+
+    public init(
+        userInputReader: UserInputReading = UserInputReader(),
+        system: Systeming,
+        devEnvironment: DeveloperEnvironmenting
```

Comment:
is there a reason why these two don't have default values?

I can see why `developerEnvironment` since it needs to hold onto a state, but why `System`?

The default values for these two are not used in a couple of other places, so the same goes in the other instances.

Reply:
I think we could adjust these callsites to use default parameters.  The only case i could think of is when we would want to both store an input property and pass it to the `SimulatorController` initializer, (e.g. if an input to `TargetBuilder` also went into `SimulatorController`) but I dont see any usages like that.

Reply:
Changed that initializer of `SimulatorController` to have defaults